internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -190,7 +190,13 @@ func (f *FleetGateway) doExecute(ctx context.Context, bo backoff.Backoff) (*flee
 			}
 
 			if !bo.Wait() {
-				// Something bad has happened and we log it and we should update our current state.
+				// One reason this returns false is when the context is cancelled
+				if ctx.Err() != nil {
+					return nil, ctx.Err()
+				}
+
+				// This should not really happen, but just in-case this error is used to show that
+				// something strange occurred and we want to log it and report it.
```

Comment:
Nit: in case the context is cancelled I guess we could break the loop and have a single place where we handle it (see line 230 below)
```suggestion
				if ctx.Err() != nil {
					// if the context is cancelled, break out of the loop
					break
				}

				// This should not really happen, but just in-case this error is used to show that
				// something strange occurred and we want to log it and report it.
```

Reply:
Done.